FarihaIS/VanRoomies
# feat(chat integration): Integrate Frontend Chat Fragment with Backend

# Welcome to VanRoomies! 👋✈️

Fixes: #18 
Fixes: #50 

## Description of the change:
Integrates the frontend chat page with the backend. Currently, the chat fragment page shows a list of chats currently open with other users. The preview includes their profile image and name. The block/delete button is currently not functional. This will be looked into with #63 and #64.

Clicking on the chat preview opens up a chat channel with the other user. A message can be typed in the prompt box at the bottom and upon clicking 'Send', the message should be added to the list of messages between the users.

The matches fragment is partially done. It needs to be integrated and the right action needs to be taken when a match is swiped left or right. This will be looked into with #49 and #65.

Real-time notifications are still in the works and are not yet implemented. This will be looked into with #66.

## How to manually test:
1. Start up MongoDB
2. Run `npm run dev` inside the backend folder (At the time of reviewing this PR, if https://github.com/FarihaIS/VanRoomies/pull/45 is still open, you will need to checkout to the `be-chatting` branch, pull, and run this command on that branch)
3. Go to Postman and run `POST Create New User` twice to create two users
4. Take the outputted `_id`s and run `POST Create New Conversation` with the `_id`s inserted in the `users` field
5. Populate the `messages` field with a list of strings representing texts. 
6. Repeat steps 4 and 5 as many times as you'd like
7. Take either of the `_id`s from before and paste it into `ChatFragment` in the beginning of `onCreateView`. The `thisUserId` field is hardcoded at the moment.
8. Build and run the app
9. The chats should show up in the chat fragment as well as the chat channel after clicking on a specific chat!

Note that after sending a chat, going back to the chat fragment, then clicking on the chat, the recent texts may not show up. You will have to click on another fragment then click back on the chat fragment and then the individual chat. This is because of #62, which needs to be explored further.
